### PR TITLE
Add book format 456

### LINKED ISSUE #456: Add Book format
**Describe the solution you'd like**

A user may want to keep track of different types of books. The `Book` entity should have a `BookFormat` enum as a member. For now, the `BookFormat` enum should contain the following values:

- Hardback
- Paperback
- eBook

---
If you need a hand with this, feel free to get in touch with us over our [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ) workspace

### PR DESCRIPTION
**Pull request by GxPinheiro**
## Summary of change

Adding BookFormat  to BookEntity and creating BookFormat ENUM

## Related issue

Closes #456

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw)

### CODE REVIEW

**Comment by knjk04** [comment]
@GxPinheiro Thanks for working on this! We'll review this shortly!

File: package-lock.json [hunk 1]
```diff
@@ -9875,4 +9875,4 @@
       }
     }
   }
-}
+}
```
Comment (knjk04) [inline package-lock.json:9878]: Could you please undo this change?

File: src/main/java/com/karankumar/bookproject/backend/entity/BookFormat.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+
+package com.karankumar.bookproject.backend.entity;
+
+public enum BookFormat {
+    // This should be kept in alphabetical order
+    HARDBACK("Hardback"),
+    PAPERBACK("Paperback"),
+    EBOOK("eBook");
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/backend/entity/BookFormat.java:25]: Could you please ensure this is in alphabetical order? It's not an issue now, but if we add more later, it'll help

File: src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java [hunk 3]
```diff
@@ -18,14 +18,8 @@
...
 import com.karankumar.bookproject.annotations.IntegrationTest;
-import com.karankumar.bookproject.backend.entity.Tag;
-import com.karankumar.bookproject.backend.utils.PredefinedShelfUtils;
-import com.karankumar.bookproject.backend.entity.Author;	
-import com.karankumar.bookproject.backend.entity.Book;	
-import com.karankumar.bookproject.backend.entity.BookGenre;	
-import com.karankumar.bookproject.backend.entity.CustomShelf;	
-import com.karankumar.bookproject.backend.entity.PredefinedShelf;	
-import com.karankumar.bookproject.backend.entity.RatingScale;
+import com.karankumar.bookproject.backend.entity.*;
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java:21]: Please note that we are not using wildcard imports

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for adding the book format! I've left some feedback

**Comment by knjk04** [comment]
Also, could you please resolve the merge conflict?

File: src/main/resources/db/migration/V003__CREATE_BOOK.sql [hunk 4]
```diff
@@ -7,6 +7,7 @@ CREATE TABLE book
     date_started_reading  DATE,
     edition               VARCHAR(255),
     book_genre            INTEGER,
+    book_format           INTEGER,
```
Comment (knjk04) [inline src/main/resources/db/migration/V003__CREATE_BOOK.sql:10]: Instead of modifying this migration, could you please create a new migration for this (for an example, see v14 on master)?

Comment (GxPinheiro) [reply src/main/resources/db/migration/V003__CREATE_BOOK.sql:10]: Alright will review all this

**Comment by knjk04** [comment]
@GxPinheiro I noticed you added another commit. When you're ready for another review, let us know (please also resolve the merge conflict)

**Comment by GxPinheiro** [comment]
> @GxPinheiro I noticed you added another commit. When you're ready for another review, let us know (please also resolve the merge conflict)

Just fixed all the points and the conflict, all ready for review

**Review by knjk04 — APPROVED** [review]
All good! Thanks for working on this!
